test/notifications/display_test.dart

```diff
@@ -495,6 +534,112 @@ void main() {
+      await init();
+      final message = eg.streamMessage();
+      final data = messageFcmMessage(message);
+      final expectedGroupKey = '${data.realmUri}|${data.userId}';
+
+      check(testBinding.androidNotificationHost.activeNotifications).isEmpty();
+      await receiveFcmMessage(async, data);
+      check(testBinding.androidNotificationHost.activeNotifications).deepEquals(<Condition<Object?>>[
+        conditionActiveNotif(data, 'stream:${message.streamId}:${message.topic}'),
+        conditionSummaryActiveNotif(expectedGroupKey),
+      ]);
+      testBinding.firebaseMessaging.onMessage.add(
+        RemoteMessage(data: removeFcmMessage([message]).toJson()));
+      async.flushMicrotasks();
+      check(testBinding.androidNotificationHost.activeNotifications).isEmpty();
+
+      check(testBinding.androidNotificationHost.activeNotifications).isEmpty();
+      await receiveFcmMessage(async, data);
+      check(testBinding.androidNotificationHost.activeNotifications).deepEquals(<Condition<Object?>>[
+        conditionActiveNotif(data, 'stream:${message.streamId}:${message.topic}'),
+        conditionSummaryActiveNotif(expectedGroupKey),
+      ]);
+      testBinding.firebaseMessaging.onBackgroundMessage.add(
+        RemoteMessage(data: removeFcmMessage([message]).toJson()));
+      async.flushMicrotasks();
+      check(testBinding.androidNotificationHost.activeNotifications).isEmpty();
+    })));
+
+    test('remove: clears conversation only if the removal event is for the last message', () => runWithHttpClient(() => awaitFakeAsync((async) async {
+      await init();
+      final stream = eg.stream();
+      const topicA = 'Topic A';
+      final message1 = eg.streamMessage(stream: stream, topic: topicA);
+      final data1 = messageFcmMessage(message1, streamName: stream.name);
+      final message2 = eg.streamMessage(stream: stream, topic: topicA);
+      final data2 = messageFcmMessage(message2, streamName: stream.name);
+      final message3 = eg.streamMessage(stream: stream, topic: topicA);
+      final data3 = messageFcmMessage(message3, streamName: stream.name);
+      final expectedGroupKey = '${data1.realmUri}|${data1.userId}';
+
+      check(testBinding.androidNotificationHost.activeNotifications).isEmpty();
+
+      await receiveFcmMessage(async, data1);
+      await receiveFcmMessage(async, data2);
+      await receiveFcmMessage(async, data3);
+
+      check(testBinding.androidNotificationHost.activeNotifications).deepEquals(<Condition<Object?>>[
+        conditionActiveNotif(data3, 'stream:${stream.streamId}:$topicA'),
+        conditionSummaryActiveNotif(expectedGroupKey),
+      ]);
+
+      testBinding.firebaseMessaging.onMessage.add(
+        RemoteMessage(data: removeFcmMessage([message1, message2]).toJson()));
+      async.flushMicrotasks();
+
+      check(testBinding.androidNotificationHost.activeNotifications).deepEquals(<Condition<Object?>>[
```

Comment:
This could benefit from a brief comment telling the story of the test:
```suggestion
      // A RemoveFcmMessage for the first two messages; the notification stays.
      testBinding.firebaseMessaging.onMessage.add(
        RemoteMessage(data: removeFcmMessage([message1, message2]).toJson()));
      async.flushMicrotasks();

      check(testBinding.androidNotificationHost.activeNotifications).deepEquals(<Condition<Object?>>[
```

(and then a couple of nits above apply to other aspects of these lines)

Similarly later in this test case:
```dart
      // Then a RemoveFcmMessage for the last message; clear the notification.
```

The test name above already gives a good summary of the plot. But I think these are helpful to make sure the reader sees the point, and how the details connect to the idea that's being tested.